### PR TITLE
Teaching assistants tests

### LINKED ISSUE #210: Teaching assistant code coverage
Create the following workflows for the teaching assistant:
* Good Workflow
    * See only assignments where you are a TA
    * Add/Modify/Delete assignment for the courses where you are a TA
    * Download homework for review (also check if TA for that course)
    * Add/Modify/Delete Review (it also checks the Grade - also check if TA for that course)

* Bad Workflow:
    * Try to change submission where you are not a TA - should return error

### PR DESCRIPTION
**Pull request by gmuraru**
Fixes #210 
Add tests for the teaching assistants' workflows.

### CODE REVIEW

File: testsuite/test_student_workflow.py [hunk 1]
```diff
@@ -14,7 +14,6 @@
 FILEPATH = settings.BASE_DIR / 'testsuite' / 'test.zip'
 
 
-@pytest.mark.django_db
```
Comment (Jokeswar) [inline testsuite/test_student_workflow.py:17]: Q: Why do we need this to be removed?

File: testsuite/test_ta_workflow.py [hunk 2]
```diff
@@ -0,0 +1,776 @@
+from datetime import datetime, timezone
+
+import pytest
+
```
Comment (Jokeswar) [inline testsuite/test_ta_workflow.py:4]: No new line

File: testsuite/test_ta_workflow.py [hunk 3]
```diff
@@ -0,0 +1,776 @@
...
+    form = MockObject()
+    form.changed_data = ["teaching_assistants"]
+    form.cleaned_data = {"teaching_assistants": [user, ta]}
+
+    test_course_admin = CourseAdmin(model=Course, admin_site=AdminSite())
+    test_course_admin.save_model(
+                obj=course,
+                request=request,
+                form=form,
+                change=True)
```
Comment (Jokeswar) [inline testsuite/test_ta_workflow.py:42]: The parenthesis should be on the next line and `True` should be followed by a final comma.

File: testsuite/test_ta_workflow.py [hunk 4]
```diff
@@ -0,0 +1,776 @@
...
+        'course': course.pk,
+        'name': 'a_new',
+        'max_score': 100,
+        'deadline_soft_0': date.date(),
+        'deadline_soft_1': date.time(),
+        'deadline_hard_0': date.date(),
+        'deadline_hard_1': date.time(),
+        'min_time_between_uploads': 30,
+        'language': 'c',
+        'value': 'Save'
```
Comment (Jokeswar) [inline testsuite/test_ta_workflow.py:173]: Final comma

File: testsuite/test_ta_workflow.py [hunk 5]
```diff
@@ -0,0 +1,776 @@
...
+    }
+    response = client.post(
+        f'/admin/interface/assignment/add/',
+        data=assignment_params
```
Comment (Jokeswar) [inline testsuite/test_ta_workflow.py:177]: Final comma

File: testsuite/test_ta_workflow.py [hunk 6]
```diff
@@ -0,0 +1,776 @@
...
+    )
+
+    STC.assertRedirects(
+        response,
+        '/admin/interface/assignment/'
```
Comment (Jokeswar) [inline testsuite/test_ta_workflow.py:182]: Final comma

File: testsuite/test_ta_workflow.py [hunk 7]
```diff
@@ -0,0 +1,776 @@
...
+        'deadline_soft_1': date.time(),
+        'deadline_hard_0': date.date(),
+        'deadline_hard_1': date.time(),
+        'min_time_between_uploads': 30,
+        'language': 'c',
+        'value': 'Save'
+    }
+    response = client.post(
+        f'/admin/interface/assignment/add/',
+        data=assignment_params
```
Comment (Jokeswar) [inline testsuite/test_ta_workflow.py:224]: Final commas

File: testsuite/test_ta_workflow.py [hunk 8]
```diff
@@ -0,0 +1,776 @@
...
+    }
+
+    response = client.post(
+        f'/admin/interface/assignment/{assignment.pk}/change/',
+        data=assignment_change_params
+    )
+
+    STC.assertRedirects(
+        response,
+        '/admin/interface/assignment/'
```
Comment (Jokeswar) [inline testsuite/test_ta_workflow.py:269]: Final commas

File: testsuite/test_ta_workflow.py [hunk 9]
```diff
@@ -0,0 +1,776 @@
...
+        'min_time_between_uploads': assignment.min_time_between_uploads,
+        'language': assignment.language,
+        'value': 'Save'
+
+    }
+
+    response = client.post(
+        f'/admin/interface/assignment/{assignment.pk}/change/',
+        data=assignment_change_params,
+        follow=True
```
Comment (Jokeswar) [inline testsuite/test_ta_workflow.py:322]: Final commas

File: testsuite/test_ta_workflow.py [hunk 10]
```diff
@@ -0,0 +1,776 @@
...
+    client.login(username=ta.username, password='pw')
+
+    response = client.post(
+        f'/admin/interface/assignment/{assignment.pk}/delete/',
+        data={"post": "yes"}
+    )
+
+    STC.assertRedirects(
+        response,
+        '/admin/interface/assignment/'
```
Comment (Jokeswar) [inline testsuite/test_ta_workflow.py:353]: Final commas

File: testsuite/test_ta_workflow.py [hunk 11]
```diff
@@ -0,0 +1,776 @@
...
+        deadline_soft=datetime(2050, 1, 1, tzinfo=timezone.utc),
+        deadline_hard=datetime(2050, 1, 1, tzinfo=timezone.utc),
+        repo_url='https://github.com/vmck/assignment',
+        repo_path='pc-00',
+    )
+
+    response = client.post(
+        f'/admin/interface/assignment/{assignment.pk}/delete/',
+        data={"post": "yes"},
+        follow=True
```
Comment (Jokeswar) [inline testsuite/test_ta_workflow.py:382]: Final comma

File: testsuite/test_ta_workflow.py [hunk 12]
```diff
@@ -0,0 +1,776 @@
...
+            course_admin._add_new_ta(ta)
+            course.teaching_assistants.set([ta])
+
+    client.login(username=ta.username, password='pw')
+    response = client.get('/admin/interface/assignment/add/')
+    assert response.status_code == 200
+
+    found_courses = set(map(
+        lambda x: x[1][0]['label'],
+        response.context['widget']['optgroups']
```
Comment (Jokeswar) [inline testsuite/test_ta_workflow.py:425]: Final comma

File: testsuite/test_ta_workflow.py [hunk 13]
```diff
@@ -0,0 +1,776 @@
...
+        deadline_soft=datetime(2050, 1, 1, tzinfo=timezone.utc),
+        deadline_hard=datetime(2050, 1, 1, tzinfo=timezone.utc),
+        repo_url='https://github.com/vmck/assignment',
+        repo_path='pc-00',
+    )
+
+    submission = assignment.submission_set.create(
+        score=100.00,
+        state=Submission.STATE_DONE,
+        user=user
```
Comment (Jokeswar) [inline testsuite/test_ta_workflow.py:474]: Final comma

File: testsuite/test_ta_workflow.py [hunk 14]
```diff
@@ -0,0 +1,776 @@
...
+        score=100.00,
+        state=Submission.STATE_DONE,
+        user=user,
+    )
+
+    client.login(username=ta.username, password='pw')
+
+    response = client.post(
+        f'/submission/{submission.pk}/rerun',
+        follow=True
```
Comment (Jokeswar) [inline testsuite/test_ta_workflow.py:586]: Final comma

File: testsuite/test_ta_workflow.py [hunk 15]
```diff
@@ -0,0 +1,776 @@
...
```
Comment (Jokeswar) [inline testsuite/test_ta_workflow.py:620]: Final comma

File: testsuite/test_ta_workflow.py [hunk 16]
```diff
@@ -0,0 +1,776 @@
...
+    client.login(username=ta.username, password='pw')
+    submission = assignment.submission_set.create(
+        score=100.00,
+        state=Submission.STATE_DONE,
+        user=user,
+    )
+
+    response = client.post(
+        f'/submission/{submission.pk}/recompute',
+        follow=True
```
Comment (Jokeswar) [inline testsuite/test_ta_workflow.py:643]: Final comma

File: testsuite/test_ta_workflow.py [hunk 17]
```diff
@@ -0,0 +1,776 @@
...
+        score=100.00,
+        state=Submission.STATE_DONE,
+        user=user,
+    )
+
+    client.login(username=ta.username, password='pw')
+
+    response = client.post(
+        f'/submission/{submission.pk}/recompute',
+        follow=True
```
Comment (Jokeswar) [inline testsuite/test_ta_workflow.py:678]: Final comma

File: testsuite/test_ta_workflow.py [hunk 18]
```diff
@@ -0,0 +1,776 @@
...
+
+    client.login(username=ta.username, password='pw')
+
+    response = client.post(
+        f'/admin/interface/assignment/',
+        data={
+            'action': 'download_review_submissions',
+            '_selected_action': '1'
+        },
+        follow=True
```
Comment (Jokeswar) [inline testsuite/test_ta_workflow.py:701]: Final commas

File: testsuite/test_ta_workflow.py [hunk 19]
```diff
@@ -0,0 +1,776 @@
...
+
+    client.login(username=ta.username, password='pw')
+
+    response = client.post(
+        f'/admin/interface/assignment/',
+        data={
+            'action': 'download_all_submissions',
+            '_selected_action': '1'
+        },
+        follow=True
```
Comment (Jokeswar) [inline testsuite/test_ta_workflow.py:725]: Final commas

File: testsuite/test_ta_workflow.py [hunk 20]
```diff
@@ -0,0 +1,776 @@
...
+
+    client.login(username=ta.username, password='pw')
+
+    response = client.post(
+        f'/admin/interface/assignment/',
+        data={
+            'action': 'run_moss',
+            '_selected_action': '1'
+        },
+        follow=True
```
Comment (Jokeswar) [inline testsuite/test_ta_workflow.py:748]: Final commas

File: testsuite/test_ta_workflow.py [hunk 21]
```diff
@@ -0,0 +1,776 @@
...
+    assert response.status_code == 200
+    assert response.json()['type'] == "run_moss"
+
+
+@pytest.mark.django_db
+def test_ta_login(client, STC, base_db_setup):
+    (_, ta, _, _, _) = base_db_setup
+    client.post(
+        '/',
+        {'username': ta.username, 'password': 'pw'}
```
Comment (Jokeswar) [inline testsuite/test_ta_workflow.py:760]: Final comma

File: testsuite/test_ta_workflow.py [hunk 22]
```diff
@@ -0,0 +1,776 @@
...
+    )
+
+    response = client.post(
+        '/admin/'
```
Comment (Jokeswar) [inline testsuite/test_ta_workflow.py:764]: This would look better as a single line

File: testsuite/test_student_workflow.py (hunk 1 above)
Comment (gmuraru) [reply testsuite/test_student_workflow.py:17]: Mistake, I will put it back

File: testsuite/test_ta_workflow.py (hunk 21 above)
Comment (gmuraru) [reply testsuite/test_ta_workflow.py:760]: Is there a comma required here if I put it on the same line?

**Pull request merged**